log/logtest/config.go

```diff
@@ -0,0 +1,50 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package logtest // import "go.opentelemetry.io/otel/log/logtest"
+
+import (
+	"context"
+
+	"go.opentelemetry.io/otel/log"
+)
+
+var defaultEnabledFn = func(context.Context, log.Record) bool {
+	return true
+}
+
+type config struct {
+	enabledFn enabledFn
+}
+
+func newConfig(options []Option) config {
+	var c config
+	for _, opt := range options {
+		c = opt.apply(c)
+	}
+
+	if c.enabledFn == nil {
+		c.enabledFn = defaultEnabledFn
+	}
+
+	return c
+}
+
+// Option configures a [Recorder].
+type Option interface {
+	apply(config) config
+}
+
+type optFunc func(config) config
+
+func (f optFunc) apply(c config) config { return f(c) }
+
+// WithEnabledFn allows configuring whether the recorder enables specific log entries or not.
+//
+// By default, every log entry will be enabled.
+func WithEnabledFn(fn enabledFn) Option {
```

Comment:
nit: I would rename it to `WithEnabledFunc` as it resembled signatures like `http.HandleFunc`